### DISCUSSION TITLE
Bootstrap a workspace

### DISCUSSION THREAD

**Discussion by tobiaskaestner** [coder/coder#11067] [General] [answered]
Hello,

my use case involves populating a workspace with a lot of external packages (tar-balled for the sake of the discussion) that need to be downloaded to the persistent volume once only. If I understood the terraform resources correctly simply downloading the stuff in the startup_script would re-download the tarballs whenever a workspace would be stopped and later restarted, right?

Is there a canonical way how to solve this with Coder/Terraform? 

Apologies if this is a noob question ...

**Comment by matifali** [comment]
Why don't you use a pre-build image with all things needed already present?

**Reply by tobiaskaestner** [reply]
I thought about that, but didn't want to include the git repositories into the image as this is mixing the tools (docker image) with the actual code ...
also, thought this might be a common use case with terraform/coder and just didn't know about the obvious way myself

**Reply by matifali** [reply, marked as answer]
I understand. You could probably use a [`coder_script`](https://registry.terraform.io/providers/coder/coder/latest/docs/resources/script) resource where you can clone/download the repos/packages conditionally by checking if they do not exist already.

You can also use our [`git-clone`](https://registry.coder.com/modules/git-clone) module in your templates to clone each of your git repos. It will skip cloning if a repo directory already exists.
